From 5d1a4a4d8ebe63673fee6ce4db32b916b48af7ef Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Tue, 14 Jan 2014 11:16:42 +0800
Subject: [PATCH 54/72] ENGR00293660 RTSP:MPEG4 streams cannot be played

confirm string is not null before calling strcmp on it

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 media/libstagefright/ACodec.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index b5e0aff..4f4538e 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -1139,12 +1139,15 @@ status_t ACodec::configureCodec(
         if (err == OK) {
             AString temp;
             if (!strcasecmp(MEDIA_MIMETYPE_VIDEO_AVC, mime)) {
-                CHECK(msg->findString("disreorder", &temp));
+                msg->findString("disreorder", &temp);
             }
 
             DisableAVCReorderParams params;
             InitOMXParams(&params);
-            params.bDisable = (!strcmp(temp.c_str(), "1")) ? OMX_TRUE : OMX_FALSE;
+            if(!temp.c_str())
+                params.bDisable = OMX_FALSE;
+            else
+                params.bDisable = (!strcmp(temp.c_str(), "1")) ? OMX_TRUE : OMX_FALSE;
             ALOGI("Send reorder config(%d) to VPU",params.bDisable);
             err = mOMX->setParameter(
                     mNode, index, &params, sizeof(params));
-- 
1.8.0

